### ISSUE TITLE
Sqlsmith: Generates mview with duplicate columns

### ISSUE THREAD

**Issue by kwannoel** [singularity-data/risingwave#4069]
**Describe the bug**
Currently mview is sometimes generated with duplicated columns:
```
thread 'main' panicked at 'duplicated column name m0.bid._row_id#1 in table m6 ', src/frontend/src/catalog/table_catalog.rs:185:21
```

**To Reproduce**
Not sure how yet...
Q: What queries could potentially bypass frontend checks, is this due to `_row_id` being implicitly selected?

**Expected behavior**
No duplicate columns should be generated.
Additionally, this should be caught during previous phases?

**Additional context**
https://buildkite.com/singularity-data/main/builds/864#01821ef1-20eb-46fb-8605-d0a4e478c7ab

**Comment by kwannoel** [comment]
Reproduced by doing heavier load test on `mview`:
```sh
thread 'run_sqlsmith_on_frontend_15' panicked at 'duplicated column name m80.m68.m29.col_3#1 in table m132 ', src/frontend/src/catalog/table_catalog.rs:185:21
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
thread 'run_sqlsmith_on_frontend_15' panicked at 'Panic while running SQL: CREATE MATERIALIZED VIEW m132 AS SELECT 35.90471585942896 AS col_0, REAL '85.8752154994714' AS col_1, FLOAT '67.51435155350298' AS col_2, TIMESTAMP '2022-07-11 00:52:42' AS col_3 FROM m80 AS t_0 JOIN m80 AS t_1 ON t_0.col_3 = t_1.col_3
```

```sql
CREATE MATERIALIZED VIEW m132 AS 
SELECT 
  35.90471585942896 AS col_0, 
  REAL '85.8752154994714' AS col_1, 
  FLOAT '67.51435155350298' AS col_2, 
  TIMESTAMP '2022-07-11 00:52:42' AS col_3 
FROM 
  m80 AS t_0 
  JOIN m80 AS t_1 ON t_0.col_3 = t_1.col_3
```

**Comment by kwannoel** [comment]
Closing this as it is not exactly a Sqlsmith bug, will open a separate issue to track this properly.
That being said, the takeaway for the current port of Sqlsmith is that the generation of column names, relation names, management of context is too ambiguous.
I realized when debugging that I'm not immediately certain whether such a query should or should not be generated by Sqlsmith.
I will come up with a proper design for it.
